chat-plugins/room-events.js

```diff
@@ -14,64 +14,83 @@ exports.commands = {
 	roomevent: 'roomevents',
 	roomevents: {
 		'': function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
-			if (!room.events || Object.keys(room.events).length === 0) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!room.events || !Object.keys(room.events).length) {
 				return this.errorReply("There are currently no planned upcoming events for this room.");
 			}
 			if (!this.runBroadcast()) return;
 			let buff = '<table border="1" cellspacing="0" cellpadding="3">';
 			buff += '<th>Event Name:</th><th>Event Description:</th><th>Event Date:</th>';
-			let events = Object.keys(room.events);
-			for (let i = 0; i < events.length; i++) {
-				buff += `<tr><td>${Chat.escapeHTML(events[i])}</td><td>${room.events[events[i]].desc}</td><td>${Chat.escapeHTML(room.events[events[i]].date)}</td></tr>`;
+			for (let i in room.events) {
+				buff += `<tr><td>${Chat.escapeHTML(room.events[i].eventName)}</td><td>${Chat.parseText(room.events[i].desc)}</td><td>${Chat.escapeHTML(room.events[i].date)}</td></tr>`;
```

Comment:
I'd rather keep the string syntax consistent, and not use concatenations together with template strings